src/dotnet/APIView/APIView/Model/TokenTreeModel.cs

```diff
@@ -0,0 +1,182 @@
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace csharp_api_parser.TreeToken
+{
+    public enum StructuredTokenKind
+    {
+        Content = 0,
+        LineBreak = 1,
+        NoneBreakingSpace = 2,
+        TabSpace = 3,
+        ParameterSeparator = 4,
+        Url = 5
+    }
+
+
+    public class StructuredToken
+    {
+        public string Value { get; set; } = string.Empty;
+        public string Id { get; set; }
+        public StructuredTokenKind Kind { get; set; }
+        public HashSet<string> Tags { get; set; } = new HashSet<string>();
```

Comment:
Are we using Tags  token for annotations?

Reply:
Not sure what you mean by Tags tokens. The tags just captures opt in or opt out properties of the token.

Reply:
Can you document all the options supported by tag?

Reply:
I have done that